csml_engine/src/lib.rs

```diff
@@ -88,6 +91,88 @@ pub fn get_open_conversation(client: &Client) -> Result<Option<DbConversation>,
     get_latest_open(client, &mut db)
 }
 
+/**
+ * create bot state
+ * 
+ * { "statusCode": i32, "body": version_id }
+ */
+pub fn create_bot_version(csml_bot: CsmlBot) -> Result<String, EngineError> {
+    let mut db = init_db()?;
+
+    let bot_id = csml_bot.id.clone();
+
+    match validate_bot(csml_bot.clone()) {
+        CsmlResult {
+            errors: Some(errors),
+            ..
+        } => Err(EngineError::Interpreter(format!("{:?}", errors))),
+        CsmlResult { .. } => bot::create_bot_version(bot_id, csml_bot, &mut db),
+    }
+}
+
+/**
+ * get by bot_id
+ * {"statusCode": i32, "body": {"bot": bot, "version_id": String}}
```

Comment:
flatten bot + version_id in one object!

Reply:
this function should not return a statusCode
it should however return { id, version_id, flows, default_flow, name, ... } as a single flat object
Then the server should wrap it with the appropriate statusCode and handle the errors